meta/ColorMathInterface.m

```diff
@@ -0,0 +1,422 @@
+       rule
+    },
+       rule = Switch[{inPCCharge, VCCharge, SCCharge, FCCharge},
+          {T, S, T, S}, {
+             FlipDeltaIdxRule[inP, SIn],
+             FlipDeltaIdxRule[Sout, outP],
+             FlipDeltaIdxRule[SIn, Sout]
+          },
+          {T, S, S, T}, {
+             FlipDeltaIdxRule[inP, Fin],
+             FlipDeltaIdxRule[Fout, outP]
+          },
+          (* closed color loop *)
+          {S, S, T, T}, {
+             FlipDeltaIdxRule[SIn, Fin],
+             FlipDeltaIdxRule[Fout, Sout],
+             FlipDeltaIdxRule[Sout, SIn],
+             FlipDeltaIdxRule[Fin, Fout]
+          },
+          {T, O, S, T}, {
+             FlipDeltaIdxRule[inP, Fin],
+             FlipDeltaIdxRule[Fout, outP]
+          },
+          {T, O, T, S},  {
+             FlipDeltaIdxRule[inP, SIn],
+             FlipDeltaIdxRule[Sout, outP],
+             (* this is fishy *)
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {SIn, Sout})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Sout, SIn})]
+          },
+          {T, O, T, O}, {
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {SIn, Fin})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Fin, SIn})],
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Fout, Sout})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Sout, Fout})],
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {SIn, Sout})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Sout, SIn})],
+             ColorMath`CMf[GetFieldColorIndex /@ {Fin, Fout, V}] -> ColorMath`CMf[GetFieldColorIndex /@ {Fout, Fin, V}]
+          },
+         {O, O, T, T}, {
+             FlipDeltaIdxRule[Sout, SIn],
+             FlipDeltaIdxRule[Fin, Fout],
+             ColorMath`CMt[{GetFieldColorIndex[inP]}, Sequence @@ (GetFieldColorIndex /@ {SIn, Fin})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[inP]}, Sequence @@ (GetFieldColorIndex /@ {Fin, SIn})],
+             ColorMath`CMt[{GetFieldColorIndex[outP]}, Sequence @@ (GetFieldColorIndex /@ {Fout, Sout})] ->
+                 ColorMath`CMt[{GetFieldColorIndex[outP]}, Sequence @@ (GetFieldColorIndex /@ {Sout, Fout})],
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Sout, SIn})] ->
+                ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {SIn, Sout})],
+             ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Fin, Fout})] ->
+                ColorMath`CMt[{GetFieldColorIndex[V]}, Sequence @@ (GetFieldColorIndex /@ {Fout, Fin})]
+         },
+          _, {}
+       ];
+       rule
+    ];
+
+(*NumberOfExternalParticles[l_List] :=*)
```

Comment:
Please consider removing the subsequent lines, as they are all commented out.

Reply:
Large comment was removed in 0b93cdc